src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```
@@ -100,51 +100,55 @@ object Boot extends App with LazyLogging {
       val statusService = new StatusService(directoryDAO, cloudExtensions, 10 seconds)
       val managedGroupService = new ManagedGroupService(resourceService, policyEvaluatorService, resourceTypeMap, accessPolicyDAO, directoryDAO, cloudExtensions, emailDomain)
 
-      val samRoutes = cloudExtensions match {
-        case googleExt: GoogleExtensions => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO, policyEvaluatorService) with StandardUserInfoDirectives with GoogleExtensionRoutes {
+      val samRoutes = (cloudExtensions, googleFirestoreOps) match {
+        case (googleExt: GoogleExtensions, Some(firestore)) => new SamRoutes(resourceService, userService, statusService, managedGroupService, config.as[SwaggerConfig]("swagger"), directoryDAO, policyEvaluatorService) with StandardUserInfoDirectives with GoogleExtensionRoutes {
```

i think from reading it that you mean "you must define both google extensions AND firestore, or neither" -- in which case maybe you want to make the case where only one of them is defined an explicit error, rather than a silent fallback to the no-extensions case?